tests/drivers/dma/scatter_gather/testcase.yaml

```diff
@@ -4,3 +4,4 @@ tests:
     tags: drivers dma
     platform_allow: intel_adsp_cavs25 intel_adsp_cavs20 intel_adsp_cavs18 intel_adsp_cavs15
      frdm_k64f mimxrt1060_evk
+    filter: dt_alias_exists("dma0")
```

Comment:
Is this really necessary given that this test specifically "whitelists" the platforms that should be tested?

If `dma0` does not exist for the whitelisted platforms, it would be better to make the test fail.

Reply:
It is as it ensure the test only builds if the requirements for the test are meet.

Not all platforms that set `dma0` alias may the test be valid for and thus platform_allow further qualifies things.

So I think its reasonable to have both.

Reply:
My only concern is that adding a new platform to the `platform_allow` list without adding the corresponding overlay will now result in a skip due to the filter, which can easily go unnoticed.

Reply:
true, but that is probably true of a larger number of samples & tests that have both filters and platform_allow.

So an enhancement to twister to report this would be useful :).

Reply:
> larger number of samples & tests that have both filters and platform_allow.

Fair enough, no strong opinion in this case.